packages/core/src/fhir-to-cda/cda-templates/components/encounters.ts

```diff
@@ -0,0 +1,296 @@
+          },
+          {
+            "#text":
+              formatDateToHumanReadableFormat(encounter.resource.period?.start) ?? NOT_SPECIFIED,
+          },
+        ],
+      },
+    },
+  ];
+}
+
+function buildAddressText(address: Address | undefined): string | undefined {
+  if (!address) return undefined;
+  return `${address.line?.join(", ")}, ${address.city}, ${address.state} ${address.postalCode}`;
+}
+
+function getPractitionerInformation(participant: Practitioner[] | undefined): string {
+  if (!participant) return NOT_SPECIFIED;
+
+  const practitionerInfo = participant
+    .map(p => buildNameText(p.name))
+    .filter(Boolean)
+    .join("\n");
+
+  return practitionerInfo ?? NOT_SPECIFIED;
+}
+
+function getLocationInformation(location: Location[] | undefined) {
+  if (!location) return undefined;
+
+  return location.map(l => {
+    return { name: l.name, address: buildAddressText(l.address) };
+  });
+}
+
+function buildNameText(names: HumanName[] | undefined): string | undefined {
+  if (!names) return undefined;
+
+  const uniqueNames = new Set<string>();
+
+  names.forEach(n => {
+    const firstAndLast = `${n.family}, ${n.given?.join(" ")}`;
+    const fullName = n.suffix?.join(" ") ? `${n.suffix?.join(" ")} ${firstAndLast}` : firstAndLast;
+    uniqueNames.add(fullName);
+  });
+
+  return Array.from(uniqueNames).join("\n");
+}
+
+function createEntryFromEncounter(
+  encounter: AugmentedEncounter,
+  referenceId: string
+): EncounterEntry {
+  const classCode = buildCodeCe({
+    code: encounter.resource.class?.code,
+    codeSystem: encounter.resource.class?.system,
+  });
+  const code = buildCodeCvFromCodeCe(classCode, encounter.resource.type);
+
+  return {
+    encounter: {
+      _classCode: "ENC",
+      _moodCode: "EVN",
+      templateId: buildInstanceIdentifier({
+        root: encounter.typeOid,
+        extension: extensionValue2015,
+      }),
+      id: buildInstanceIdentifier({
+        root: placeholderOrgOid,
+        extension: encounter.resource.id,
+      }),
+      code,
+      statusCode: {
+        _code: mapEncounterStatusCode(encounter.resource.status),
+      },
+      effectiveTime: {
+        low: withoutNullFlavorObject(
+          formatDateToCdaTimestamp(encounter.resource.period?.start),
+          "_value"
+        ),
+        high: withoutNullFlavorObject(
+          formatDateToCdaTimestamp(encounter.resource.period?.end),
+          "_value"
+        ),
+      },
+      performer: buildPerformer(encounter.practitioners),
+      participant: buildParticipant(encounter.locations),
+      entryRelationship: createEntryRelationshipObservation(encounter.resource, referenceId),
+    },
+  };
+}
+
+/**
+ * For FHIR statuses, see https://hl7.org/fhir/R4/valueset-encounter-status.html
+ * For CDA statuses, see https://terminology.hl7.org/5.2.0/ValueSet-v3-ActStatus.html
+ */
+function mapEncounterStatusCode(status: string | undefined): string {
```

Comment:
ditto: create type for Encounter status code and return it here instead of `string`?

Reply:
I think this will only complicate the function. Just let it do the string conversion and we can create the statusCode object on the caller. Otherwise, each switch case's return is going to be a complicated mess

Reply:
The caller should use whatever type we define here, not the other way around.

I'm just saying to create a type to represent EncounterStatusCode:

```ts
type EncounterStatusCode = "new" | "active" | "held" | "completed" | "nullified" | "completed";

function mapEncounterStatusCode(status: string | undefined): EncounterStatusCode {
   ...
}
```

Done.

Reply:
Ahhh, I misunderstood what you meant